orchestrator/modules/nl2sql/benchmarks/runner.py

```
@@ -79,23 +80,53 @@ async def run_benchmark(
+                # validates (read-only, LIMIT-capped) and runs under the
+                # pipeline's EXPLAIN/timeout guards — and compare result sets
+                # order-insensitively. Never again mirrored from exact match
+                # (string match is the metric the field abandoned: equivalent
+                # SQL that reads differently scored 0).
+                error_note = None
+                if execute_sql is None:
+                    exec_match = None  # honest: unmeasured without an executor
+                else:
+                    try:
+                        expected_rows = await execute_sql(expected_sql)
+                    except Exception as e:  # a broken golden is a data bug — surface it
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)